lib/ecto/changeset.ex

```diff
@@ -1347,7 +1348,7 @@ defmodule Ecto.Changeset do
   @spec validate_format(t, atom, Regex.t, Keyword.t) :: t
   def validate_format(changeset, field, format, opts \\ []) do
     validate_change changeset, field, {:format, format}, fn _, value ->
-      if value =~ format, do: [], else: [{field, {message(opts, "has invalid format"), []}}]
+      if value =~ format, do: [], else: [{field, {message(opts, "has invalid format"), [validation: :format, format: format]}}]
```

Comment:
So adding the format here is going to pose problems. That's going to break most traverse_errors implementations out there because it expects something that can be converted to a string (and a regex cannot).